tracing-subscriber/src/registry/sharded.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::bool_comparison)]
```

Comment:
Is this something where we can change the code in a minor way rather than requiring the allow attribute?

Reply:
So in this case there are several places where `== true` and `== false` were being used (e.g., https://github.com/tokio-rs/tracing/blob/a19483070c7a64440f642bb5384ead78271fb8b6/tracing-subscriber/src/registry/sharded.rs#L509-L510). Since that's always unnecessary, I thought it was probably like that for stylistic reasons, which is why I overrode the lint globally. If you don't mind changing those parts, we could remove the allow.

Reply:
I'm fine with changing the assertions. Either it seemed clearer to me at the time, or I wasn't thinking through it clearly. :)